lib/ssl/src/ssl_gen_statem.erl

```diff
@@ -1022,17 +1015,21 @@ send_alert(Alert, _, #state{static_env = #static_env{protocol_cb = Connection}}
 
 handle_own_alert(Alert0, StateName,
 		 #state{static_env = #static_env{role = Role,
+                                                 transport_cb = Transport,
+                                                 socket = Socket,
                                                  protocol_cb = Connection},
-                        ssl_options = #{log_level := LogLevel}} = State) ->
+                        ssl_options = #{log_level := LogLevel} = Opts} = State) ->
+    Alert = Alert0#alert{role = Role},
+    maybe_store_src(maps:get(keep_secrets, Opts, undefined), Connection, Transport, Socket),
```

Comment:
why using src acronym? why not `maybe_store_sockname`?

Reply:
Because I wanted a name that felt more descriptive and I call it src in keylog_info-map,  but I changed it to alert_src here 
as to add context. If it is our own alert we will save this information from sockname while it is still available, and
not at the place where we want to invoke the callback for the keylogging.